examples/research_projects/sdxl_flax/README.md

```diff
@@ -0,0 +1,63 @@
+
+JAX is a high-performance numerical computation library that is well-suited for running stable diffusion models on TPU v5e. It provides a number of features that make it easy to develop and deploy stable diffusion models. Try it out for yourself:
+
+[https://huggingface.co/spaces/google/sdxl](https://huggingface.co/spaces/google/sdxl)
+
+All JAX operations are implemented in terms of operations in [XLA](https://www.tensorflow.org/xla/) - the Accelerated Linear Algebra compiler.
+
+JAX uses just-in-time (jit) compilation of JAX Python functions so it can be executed efficiently in XLA. In order to get the best performance, we must use static shapes for jitted functions, this is because JAX transforms work by tracing a function and to determine its effect on inputs of a specific shape and type. When a new shape is introduced to an already compiled function, it retriggers compilation on the new shape, which can greatly reduce performance.
+
+Machine learning workloads can be scaled across multiple devices using JAX's [pmap](https://jax.readthedocs.io/en/latest/_autosummary/jax.pmap.html), which expresses single-program multiple-data (SPMD) programs. Applying pmap to a function will compile a function with XLA, then execute in parallel on XLA devices. 
+
+Putting this all together using a Stable Diffusion XL example, running on a TPU v5e with 4 accelerators.
+
+    ```python
+    # Load model
+    pipeline, params = FlaxStableDiffusionXLPipeline.from_pretrained(
+    model_id,
+    dtype=dtype)
+
+    prompt = "a colorful photo of a castle in the middle of the forest"
+    
+    # Device count is 4, so we have len(prompt) = 4
+    prompt = [prompt] * jax.device_count()
+    
+    # prompt shape after calling prepare_inputs = (4, 2, 77) 
+    prompt = pipeline.prepare_inputs(prompt)
+    
+    # shard is a helper function to prepare inputs to pmap.
+    # shape after calling shard = (4, 1, 2, 77)
+    prompt = shard(prompt_ids)
+
+    # replicates parameters to multiple devices
+    p_params = replicate(params)
+
+    # applying pmap to the pipelines's _generate function 
+    # will compile it,then execute it in parallel on all TPUs.
+    p_generate = pmap(
+        pipeline._generate,
+        static_broadcasted_argnums=[3, 4, 5, 9]
+    )
+    images = p_generate(prompt_ids, p_params, ...)
+    ```
+
+To see the full working code, see the `Best practices` section below.
+
+To find out more about cost vs perf trade-off and when to use a TPU v5e, see this [blogpost](https://cloud.google.com/blog/products/compute/how-cloud-tpu-v5e-accelerates-large-scale-ai-inference).
+
+## Best practices for SDXL inference in JAX with TPU and TPUv5e
+
+- Let's show how to use SDXL in inference. Start with simple 10 liner for text-to-image.
+- Explain how to change number of inference steps, guidance scale, prompt engineering, ...
+- (Nice-to-have) Show Img2Img use case, show refiner
+TODO: All the code can live in `./sdxl_single.py`
+
```

Comment:
```suggestion
```

Maybe this whole section can be covered by the `## Stable Diffusion XL pipeline in JAX` section and we remove all this? Wdyt @entrpn ?